### PR TITLE
tailcfg, ipn/ipnlocal: advertise a Service when exit node DNS proxy available

### PR DESCRIPTION
**Pull request by bradfitz**
Updates #1713

### CODE REVIEW

File: tailcfg/tailcfg.go [hunk 1]
```diff
@@ -379,18 +379,48 @@ func (h *Hostinfo) CheckRequestTags() error {
...
+	//        port number that the peerapi is running on the
+	//        node's Tailscale IPv4 address.
+	//     * "peerapi6": peerapi is available on IPv6; Port is the
+	//        port number that the peerapi is running on the
+	//        node's Tailscale IPv6 address.
+	//     * "peerapi-dns": the local peerapi service supports
+	//        being a DNS proxy (when the node is an exit
+	//        node). For this service, the Port number is really
+	//        the version number of the service.
+	Proto ServiceProto // TCP or UDP
```
Comment (crawshaw) [inline tailcfg/tailcfg.go:412]: now you have a big fancy comment you don't need the trailing comment

File: tailcfg/tailcfg.go [hunk 2]
```diff
@@ -379,18 +379,48 @@ func (h *Hostinfo) CheckRequestTags() error {
...
+
+	// Port is the port number.
```
Comment (crawshaw) [inline tailcfg/tailcfg.go:414]: "...the service is listening on." ? then drop the trailing comment?

File: ipn/ipnlocal/local.go [hunk 3]
```diff
@@ -1753,15 +1753,25 @@ func (b *LocalBackend) getPeerAPIPortForTSMPPing(ip netaddr.IP) (port uint16, ok
...
-			proto = "peerapi6"
+			proto = tailcfg.PeerAPI6
 		}
 		ret = append(ret, tailcfg.Service{
 			Proto: proto,
 			Port:  uint16(pln.port),
 		})
 	}
+	switch runtime.GOOS {
+	case "linux", "freebsd", "openbsd", "illumos", "darwin":
```
Comment (crawshaw) [inline ipn/ipnlocal/local.go:1766]: I wish this weren't so far away from the implementations of HandleExitNodeDNSQuery.

**Review by crawshaw — APPROVED** [review]

File: ipn/ipnlocal/local.go (hunk 3 above)
Comment (bradfitz) [reply ipn/ipnlocal/local.go:1766]: Same, but ran out of ideas. Welcome any good ones. Ideally this is gone in a day or so?

**Pull request merged**